Core/src/main/java/com/plotsquared/core/components/ComponentPresetManager.java

```diff
@@ -181,7 +181,13 @@ public ComponentPresetManager(final @NonNull EconHandler econHandler, final @Non
             }
             allowedPresets.add(componentPreset);
         }
-        final int size = (int) Math.ceil((double) allowedPresets.size() / 9.0D);
+        final int size;
+        if (allowedPresets.isEmpty()) {
+            player.sendMessage(TranslatableCaption.of("preset.empty"));
+            return null;
+        } else {
+            size = (int) Math.ceil((double) allowedPresets.size() / 9.0D);
+        }
```

Comment:
The size does not depend on the `if` part. It would be sufficient to just do the check before the old `final int size = ...` code.